### ISSUE TITLE
Local bounding box seems to ignore overlay option

### ISSUE THREAD

**Issue by ghost** [pgf-tikz/pgf#858]
I hope that this is just me seeing ghosts, or there is a reason for this behavior. According to what I find, the `local bounding box` now includes paths that are drawn in the overlay mode. I believe that this was not the case before version 3.1.5. 

Let us look at the example

```latex
\documentclass[tikz,border=3mm]{standalone}
\begin{document}
\begin{tikzpicture}
 \begin{scope}[local bounding box=test]
  \draw (0,0) circle[radius=1cm];
  \draw[overlay] (1,2) circle[radius=1cm];
 \end{scope}
 %\draw (test.south west) rectangle (test.north east);
\end{tikzpicture}
\end{document}
```

<img width="186" alt="Screen Shot 2020-05-22 at 9 56 27 PM" src="https://user-images.githubusercontent.com/35985244/82721931-df05e180-9c76-11ea-9b4e-2c21c62212b1.png">

The result is as expected. The overlay circle at `(1,2)` gets (partially) clipped away.

However, when I *draw* the boundary of the local bounding box, i.e. when I uncomment

```latex
\draw (test.south west) rectangle (test.north east);
```

I get

<img width="180" alt="Screen Shot 2020-05-22 at 9 58 49 PM" src="https://user-images.githubusercontent.com/35985244/82721954-3441f300-9c77-11ea-99fe-762e9425c9d9.png">

and with 

```latex
\draw[overlay] (test.south west) rectangle (test.north east);
```

<img width="161" alt="Screen Shot 2020-05-22 at 9 59 47 PM" src="https://user-images.githubusercontent.com/35985244/82721969-56d40c00-9c77-11ea-8462-060e17f3e3f5.png">

That is, the `local bounding box` captures the second circle even though it is drawn in the `overlay` mode.

I do see that this behavior can make sense, yet it seems to be different from the previous behavior. So if this is now the new "correct" behavior one may want to add this information to the manual.

**Comment by hmenke** [comment]
This is a documented breaking change of PGF 3.1.5.  Quoting from the [release notes](https://github.com/pgf-tikz/pgf/releases/tag/3.1.5):

> It turned out that in `\pgf@protocolsizes` the `\pgf@path@size@hook` was executed only if the picture was actually to be placed. This led to the problem that the local bounding box was not updated correctly (#795). To correct this, the hook is now executed unconditionally.

**Comment by hmenke** [comment]
Try this:
```latex
\makeatletter
\tikzset{
  overlay/true/.code={%
    \pgf@relevantforpicturesizefalse
    \pgf@size@hookedfalse
  },
  overlay/false/.code={%
    \pgf@relevantforpicturesizetrue
    \pgf@size@hookedtrue
  },
}
\makeatother
```

**Comment by ghost** [comment]
@hmenke  Thanks! I think it would make sense to add this information also to the manual. One could write just above the code example before section **107.5 Declaring New Shapes**

> Note that the local bounding box also includes paths that are created in the overlay mode, i.e. paths that are excluded from bounding box of the `tikzpicture`. This can lead to the curious effect that the bounding box of the picture changes when one just uses nodes of the local bounding box in a path.

**Comment by hmenke** [comment]
It would make a lot more sense to revert 03b89120f0bf0908ead1033dcca82e5abb756521 and fix #795 properly.

**Comment by hmenke** [comment]
Fixed in https://github.com/pgf-tikz/pgf/commit/ca1f30e1fb33789142c306d68cc346b89f771207
